backend/witness/witness.go

```diff
@@ -356,7 +356,7 @@ func (w *witness) FromJSON(s *schema.Schema, data []byte) error {
 	// collect all secret values; if any are missing, we just deal with the public part.
 	secretValues := make([]any, 0, s.NbSecret)
 	publicOnly := false
-	if _, err := schema.Walk(instance, ptrTyp, func(leaf schema.LeafInfo, tValue reflect.Value) error {
+	if _, err := schema.Walk(nil, instance, ptrTyp, func(leaf schema.LeafInfo, tValue reflect.Value) error {
```

Comment:
well, I guess the `ToJSON` / `FromJSON` methods are not really used, but witness could do a type switch in a method to figure out the base field or something, and provide it here to avoid a nil pointer with subtle side effects ? 
Or maybe just document in `Walk` that `nil` is a OK value and is used only in `InitHook`

Reply:
Actually we already have the underlying field in `schema.Schema`, it wasn't just available. I made it public and I'm providing it here.

Reply:
Also added comment to `schema.Walk` that the field will be passed.